--- basic_ldap.py	2020-06-18 05:56:40.119884228 +0000
+++ basic_ldap.patched.py	2020-06-18 06:16:04.042267468 +0000
@@ -32,10 +32,17 @@
             domain, username = login.split('\\')
             email = ''
             email_verified = False
-        else:
+        elif '@' in login:
             username, domain = login.split('@')
             email = login
             email_verified = True
+        else:
+            try:
+              username, domain = login, current_app.config['LDAP_DEFAULT_DOMAIN']
+              email = ''
+              email_verified = False
+            except KeyError:
+              raise ApiError('LDAP_DEFAULT_DOMAIN has to be defined', 401)
     except ValueError:
         raise ApiError('expected username with domain', 401)
 
@@ -59,7 +66,7 @@
     if not email_verified:
         try:
             ldap_result = ldap_connection.search_s(userdn, ldap.SCOPE_SUBTREE, '(objectClass=*)', ['mail'])
-            email = ldap_result[0][1]['mail'][0].decode(sys.stdout.encoding)
+            email = ldap_result[0][1]['mail'][0].decode(sys.stdout.encoding).lower()
             email_verified = True
         except Exception:
             email = '{}@{}'.format(username, domain)
